drivers/scheduler/k8s/k8s.go

```diff
@@ -4333,17 +4334,18 @@ func (k *K8s) GetVolumes(ctx *scheduler.Context) ([]*volume.Volume, error) {
 							want = true
 						}
 					}
-					// If volume is still not found, try the datavolumes
+					// If volume is still not found, try the datavolume for the vi
 					if !want && ownerRef.Kind == "DataVolume" {
 						// get datavolume
 						kvClient := k8sKubevirt.GetKubevirtClient()
-						dataVol, err := kvClient.CdiClient().CdiV1beta1().DataVolumes(vm.Namespace).Get(context.TODO(), pvc.Name, metav1.GetOptions{})
-						if err != nil {
-							return nil, fmt.Errorf("error getting datavolumes in namespace: %s", vm.Namespace)
-						}
-						for _, ownerRef := range dataVol.OwnerReferences {
-							if ownerRef.Name == vm.Name {
-								want = true
+						dataVol, err := kvClient.CdiClient().CdiV1beta1().DataVolumes(vm.Namespace).Get(context.TODO(), ownerRef.Name, metav1.GetOptions{})
+						if err != nil || k8serrors.IsNotFound(err) {
```

Comment:
the second condition is redundant. Did you mean to exit out when the error is something other than NotFound?